homeassistant/config_entries.py

```diff
@@ -911,6 +912,8 @@ def _abort_if_unique_id_configured(
                         self.hass.async_create_task(
                             self.hass.config_entries.async_reload(entry.entry_id)
                         )
+                if entry.source == SOURCE_IGNORE and allow_ignored_devices:
```

Comment:
We cannot simply ignore them, as this would create multiple entries with the same unique ID, instead the previous ignored entry needs to be cleaned up.

Reply:
Thanks for the feedback!

I assume you mean that we need to call the `config_entry.async_remove` method, to allow the entry to clean itself?
This makes sense, though currently it seems to be skipped anyway.
```
340        if self.source == SOURCE_IGNORE:
341            return
```

Also, `_abort_if_unique_id_configured` is non-async method, which makes it non-trivial to call and async method (`config_entry.async_remove`) from

Reply:
It can be removed via the config entries (the collection); `ConfigEntries.async_remove(entry_id)`